go/vt/vtctl/vtctl.go

```diff
@@ -3506,14 +3506,29 @@ func commandWorkflow(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.
 	if err != nil {
 		wr.Logger().Errorf("Keyspace %s not found", keyspace)
 	}
-
-	results, err := wr.WorkflowAction(ctx, workflow, keyspace, action, *dryRun)
-	if err != nil {
-		return err
-	}
-	if action == "show" || action == "listall" {
-		return nil
+	var results map[*topo.TabletInfo]*sqltypes.Result
+	if action == "tags" {
+		tags := ""
+		if subFlags.NArg() == 3 {
+			tags = strings.ToLower(subFlags.Arg(2))
+		}
+		results, err = wr.WorkflowTagAction(ctx, keyspace, workflow, tags)
```

Comment:
I think we should either allow multiple tags like `<ks.workflow> [tags [tags ...]]` or do some extra linting here. There's a subtle problem where if I were to do `vtctlclient Workflow ks.workflow tags tag1 tag2` then we wouldn't enter the `if subflags.NArg() == 3` branch, and it would silently delete all the tags for the workflow.

Reply:
For multiple tags, I had thought of keeping it simple (as mentioned in the PR description): you would implement it by joining the tags as `tag1,tag2` before providing it to the `Workflow` command. This limits the CLI requirement to just setting the tag value, either to a single value, csv for multiple tags or empty to clear it. I can fix the validation so that it will error if more tags are provided in the command, so that there is no silent deletion.

Reply:
(just closing the loop on this): we chatted and your implementation, plus the fix for silent deletion, makes sense!